scripts/specCoverageReport.ts

```diff
@@ -0,0 +1,424 @@
+import Table from 'cli-table';
+import dox from 'dox';
+import fs from 'fs';
+import { glob } from 'glob';
+
+const ABLY_FEATURES_SPEC_URL = 'https://raw.githubusercontent.com/ably/specification/main/textile/features.textile';
+const PATHS_TO_LOOK_FOR_TESTS = ['./test', './src/platform/react-hooks'];
+const PATHS_TO_IGNORE_FOR_TESTS = ['./test/common/ably-common'];
+
+enum SpecCoverageType {
+  spec = 'spec',
+  specpartial = 'specpartial',
+  nospec = 'nospec',
+  specskip = 'specskip',
+}
+
+interface SpecItem {
+  level: number;
+  key: string;
+  description: string;
+}
+
+interface TestSpecCoverage {
+  coverageType: SpecCoverageType.spec | SpecCoverageType.specpartial | SpecCoverageType.nospec;
+  /**
+   * Spec item key covered by the test, if applicable based on {@link coverageType} value.
+   */
+  specKey?: string;
+  /**
+   * An optional comment for tagged spec item. Looks like this in test docstrings:
+   *
+   * `@spec ID123 - [comment]`
+   */
+  coverageDescription?: string;
```

Comment:
Might be good to update the `CONTRIBUTING.md` guidance in #1806 to mention the ability to provide a comment alongside the tag?

Reply:
true, will add a format example for tags in the `CONTRIBUTING.md` in that PR

Reply:
added guidance and examples to `CONTRIBUTING.md` in https://github.com/ably/ably-js/pull/1806/commits/25c7f5cde53955fc8c7d51fb927d268739107e32 (https://github.com/ably/ably-js/pull/1806 PR)